src/if_statements.py

```diff
@@ -231,17 +231,73 @@ def add_statement(self, statement: Statement) -> None:
     def add_comment(self, contents: str) -> None:
         self.add_statement(SimpleStatement(None, comment=contents))
 
-    def add_if_else(self, if_else: IfElseStatement) -> None:
-        if if_else.if_body.ends_in_jump():
-            # Transform `if (A) { B; return C; } else { D; }`
-            # into `if (A) { B; return C; } D;`,
-            # which reduces indentation to make the output more readable
-            self.statements.append(replace(if_else, else_body=None))
-            if if_else.else_body is not None:
-                self.extend(if_else.else_body)
+    def add_if_else(self, if_else: IfElseStatement, else_jump: bool = False) -> None:
+        if if_else.else_body is not None and not if_else.if_body.ends_in_jump():
+            # Simple case; perform no further rewrites
+            self.statements.append(if_else)
             return
 
-        self.statements.append(if_else)
+        # We now know that we have an IfElseStatement like `if (A) { B; goto C; } else { D; }`
+        # where `D` may be empty. We can rewrite this into `if (A) { B; goto C; } D;`
+        # which reduces indentation to make the output more readable.
+
+        # Additionally, if `A` is a complex condition containing `&&` expressions, we could
+        # expand something like `if (A1 && A2 && A3) { B; goto C; } D;`
+        # into nested if's like `if (A1) { if (A2) { if (A3) { B; goto C; } } } D;`
+        # (This can't be done without additional gotos if there is an explicit `else` block)
+        # We perform this expansion if any of the `A1`, `A2`... subconditions are CommaConditionExprs,
+        # to avoid using the comma operator in conditions.
+
+        # Iterate through the outermost `&&` subconditions in `if_else.condition` like a zipper.
+        # We start with the whole condition in `outer_cond`, and iterate by appending the right
+        # side into `inner_cond`. At the start of each iteration, the whole condition would be
+        # represented by `join_conditions(outer_cond, "&&", inner_cond)` -- unless any conditions
+        # are removed.
```

Comment:
Would be useful to state here that outer_cond is originally structured like `(((...) && ...) && ...)` and gets converted into `(... && (... && (...)))`

Reply:
Refactored to avoid this, since it seems to affect codegen on IDO.